cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -99,10 +124,20 @@ func Run(gadgetRoot, device string, options *Options) error {
 		}
 	}
 
+	// store the encryption key as the last part of the process to reduce the
+	// possiblity of exiting with an error after the TPM provisioning
+	if options.Encrypt {
+		if err := key.Store(options.KeyFile); err != nil {
+			return err
+		}
+	}
+
```

Comment:
One thing that's missing here is provisioning the recovery key (a 128-bit key, using a second LUKS slot that's configured normally).

Reply:
A question here: why is the recovery key 128-bit while the main key is 256-bit? Shouldn't it have the same entropy as the other key?